drivers/gpio/gpio_max14906.c

```diff
@@ -0,0 +1,495 @@
+/*
+ * Copyright (c) 2024 Analog Devices Inc.
+ * Copyright (c) 2024 Baylibre SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+
+#define LOG_LEVEL CONFIG_GPIO_LOG_LEVEL
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(gpio_max14906);
+
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+#include "gpio_max14906.h"
+#include "gpio_max149x6.h"
+
+#define DT_DRV_COMPAT adi_max14906_gpio
+
+static int gpio_max14906_diag_chan_get(const struct device *dev);
+
+static int max14906_pars_spi_diag(const struct device *dev, uint8_t *rx_diag_buff, uint8_t rw)
+{
+	struct max14906_data *data = dev->data;
+	int ret = 0;
+
+	if (rx_diag_buff[0]) {
+		LOG_ERR("[DIAG] MAX14906 in SPI diag - error detected\n");
+		data->glob.interrupt.reg_bits.SHT_VDD_FAULT = 0x1 & (rx_diag_buff[0] >> 5);
+		data->glob.interrupt.reg_bits.ABOVE_VDD_FAULT = 0x1 & (rx_diag_buff[0] >> 4);
+		data->glob.interrupt.reg_bits.OW_OFF_FAULT = 0x1 & (rx_diag_buff[0] >> 3);
+		data->glob.interrupt.reg_bits.CURR_LIM = 0x1 & (rx_diag_buff[0] >> 2);
+		data->glob.interrupt.reg_bits.OVER_LD_FAULT = 0x1 & (rx_diag_buff[0] >> 1);
```

Comment:
```suggestion
#define GET_BIT(val, i)  (0x1 & ( (val) >> (i)))

		data->glob.interrupt.reg_bits.SHT_VDD_FAULT = GET_BIT(rx_diag_buff[0], 5);
		data->glob.interrupt.reg_bits.ABOVE_VDD_FAULT = GET_BIT(rx_diag_buff[0], 4);
		data->glob.interrupt.reg_bits.OW_OFF_FAULT = GET_BIT(rx_diag_buff[0], 3);
		data->glob.interrupt.reg_bits.CURR_LIM = GET_BIT(rx_diag_buff[0], 2);
		data->glob.interrupt.reg_bits.OVER_LD_FAULT = GET_BIT(rx_diag_buff[0], 1);
```

Reply:
Done.

Reply:
Thanks, could point me the definition of GET_BIT I not found it, might be missed?

Reply:
Yes , it is missing. I go the impression that `GET_BIT` already exist here.
I will add it, right now but will change to `MAX149X6_GET_BIT` since according to link you shared , it should not be generic.